### PR TITLE
Refactoring

### PR DESCRIPTION
**Pull request by rustam-tolipov**
### OOP school library: refactor your code

We have completed all the requirements:

- We have followed the rules of YAGNI, KISS, and DRY. 💯
- We have downsized the `main.rb` file. 💯

Feel free to review🧑🏻‍💻  our project and let us know if there are any changes to do. 🤓 🤓

### CODE REVIEW

File: app.rb [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+
+  def create_student
+    print 'Age: '
+    age = gets.chomp.to_i
+    print 'Name: '
+    name = gets.chomp
+    print 'Has parent permission? [Y/N]: '
+    parent_permission = gets.chomp.upcase == 'Y'
+    gets
+    @people << Student.new(name, age, parent_permission)
```
Comment (Dipeshtwis) [inline app.rb:63]: - [x] This line should've been placed in the 'create_person()' method. Check best practices and SOLID lesson [here](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/lessons/solid.md#best-practices-and-solid).

File: app.rb [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+  def create_teacher
+    print 'Age: '
+    age = gets.chomp.to_i
+    print 'Name: '
+    name = gets.chomp
+    print 'Specialization: '
+    specialization = gets.chomp
+    puts 'Person created successfully!'
+    gets
+    @people << Teacher.new(name, age, specialization)
```
Comment (Dipeshtwis) [inline app.rb:70]: - [x] This line should've been placed in the 'create_person()' method. Check best practices and SOLID lesson [here](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/lessons/solid.md#best-practices-and-solid).

**Review by Dipeshtwis — CHANGES_REQUESTED** [review]
Hi @Rustamxon7 and @ahzia, 

:+1: This is good to make it this far, your project is really nice, congratulations on the good work you have done so far.
:+1: App is working fine

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!


 ## Required Changes ♻️

**_Check the comments under the review_**

- [x] I would recommend adding the command `ruby main.rb` to run the project locally in the README file.

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

- N.A.

Cheers and Happy coding!👏👏👏
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

**_Dipesh Kumar_**


------

_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project(check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by ahzia** [comment]
@Dipeshtwis Thanks for the Review, I Implemented all required changes.

**Review by ZeenatLawal — APPROVED** [review]
Hi @Rustamxon7 and @ahzia, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 

## Congratulations! 🎉 
6vEo


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

**Pull request merged**